cmd/config/internal/commands/util.go

```diff
@@ -10,8 +10,43 @@ import (
 
 	"github.com/go-errors/errors"
 	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/cmd/config/ext"
+	"sigs.k8s.io/kustomize/kyaml/pathutil"
 )
 
+// executeCmdOnPkgs takes the function definition for a command to be executed on single package, applies that definition
+// recursively on all the subpackages present in rootPkgPath if recurseSubPackages is true, else applies the command on rootPkgPath only
+func executeCmdOnPkgs(rootPkgPath string, recurseSubPackages, needOpenAPI bool,
+	executeCmd func(string) error) error {
```

Comment:
As mentioned further up, I think we should have an interface type instead of a function type here. In any case, I think it is worth making this a named type.

Reply:
Done.